var/spack/repos/builtin/packages/py-ray/package.py

```diff
@@ -42,3 +43,15 @@ class PyRay(PythonPackage):
     depends_on('py-redis@3.3.2:3.4', type=('build', 'run'))
     depends_on('py-opencensus', type=('build', 'run'))
     depends_on('py-prometheus-client@0.7.1:', type=('build', 'run'))
+    depends_on('py-setproctitle', type=('build', 'run'))
+    depends_on('py-psutil', type=('build', 'run'))
+    depends_on('py-pickle5', type=('build', 'run'))
+
+    def setup_build_environment(self, env):
+        env.set('SKIP_THIRDPARTY_INSTALL', '1')
+
+    # Compile the dashboard npm modules included in the project
+    @run_before('install')
+    def build_dashboard(self):
+        sh = which('sh')
+        sh('-c', 'cd python/ray/dashboard/client && npm ci && npm run build')
```

Comment:
This won't work on Windows. I would do something like:
```python
with working_dir(join_path('python', 'ray', 'dashboard', 'client')):
    npm = which('npm')
    npm('ci')
    npm('run', 'build')
```

Reply:
Thanks, this is indeed much more portable. I tested it and it worked flawlessly.